### ISSUE TITLE
snap: Transition to classic confinement

### ISSUE THREAD

**Issue by jmooring** [gohugoio/hugo#11126]
The Snap package is currently subject to strict confinement. We are already approved for classic confinement. Reasons to transition to classic confinement:

1. We need to move from core20 to core22 at some point, and the current Asciidoctor/Ruby package is blocked by Apparmor under core22. See [#11116](https://github.com/gohugoio/hugo/issues/11116).
2. Allow access to projects outside of $HOME.
3. Allow access to non-standard Git config file locations without having to add more interfaces (e.g., `system-files`, `personal-files`).
4. Allow dart to access `/proc/sys/vm/max_map_count`
5. Allow access to credentials/configs used with `hugo deploy`
6. Avoid problems such as <https://discourse.gohugo.io/t/45477/>

To do:

- [ ] Create mechanism to detect if user is running Hugo as a Snap under strict confinement, then prompt then to refresh (see <https://forum.snapcraft.io/t/19892/16>). This would be in the last commit prior to the change to classic confinement.
- [ ] Decide which parts we should continue to include under classic confinement, if any:
     - Git
     - Go
     - Asciidoctor
     - Dart Sass
     - Node
     - Pandoc
     - rst2html

   I am inclined to remove _all_ of them. We _needed_ to include them under strict confinement, but under classic confinement users can install the dependencies as needed, just like any other installation method.

cc: @anthonyfok

**Comment by bep** [comment]
I know very little about how Snap works, but I vote for the "least amount of work for us" option. I'm guessing that very few people use asciidoc, and those who do can go through some extra hoops.

Is it possible to define a baseline(say Git, Go, Node, Dart Sass) and leave the rest optional?

**Comment by jmooring** [comment]
> Is it possible to define a baseline(say Git, Go, Node, Dart Sass) and leave the rest optional?

Yeah, maybe, but there are some complications that I don't understand yet. If a classically confined Snap includes Dart Sass, and the user has installed Dart Sass, which is used? Or more importantly, which _should_ be used? Further investigation is required, but there's no rush on this.

FYI: I am planning to embedded Dart Sass in the current Snap (strictly confined, core20) once things have settled a bit.

**Comment by jmooring** [comment]
> I am inclined to remove _all_ of them

So, I think I was a bit frustrated by Ruby 3.x when I wrote this. I've changed my mind.

**Comment by jmooring** [comment]
Closing for now.

We resolved the core20 to core22 transition, including the Ruby 3.x issue, in [#12227](https://github.com/gohugoio/hugo/pull/12227) with credit and many thanks to @alexmurray.

The other restrictions/problems listed in the issue description are rarely encountered in the wild, and the convenience of self-containing the required and optional dependencies far outweighs the occasional edge case.
